packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1005,7 +1146,23 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
             (success, returndata) = _contract.call{gas: _gasLimit}(_data);
         }
 
-        // Switch back to the original message context now that we're out of the call.
+        // If the message threw an exception, its value should be returned back to the sender.
+        // So, we force it back, BEFORE returning the messageContext to the previous addresses.
+        if (messageValue > 0 && !success) {
+            bool transferredOvmEth = _attemptForcedEthTransfer(
```

Comment:
Please add additional comments here as to the cases in which this transfer could fail.